catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/MgrsCoordinateProcessor.java

```diff
@@ -45,18 +45,22 @@
 public class MgrsCoordinateProcessor {
   private static final Logger LOGGER = LoggerFactory.getLogger(MgrsCoordinateProcessor.class);
 
-  private static final Pattern PATTERN_MGRS_ZONE = Pattern.compile("\\d\\d?[C-HJ-NP-X]");
-
-  private static final Pattern PATTERN_MGRS_100KM = Pattern.compile("[A-HJ-NP-Z][A-HJ-NP-V]");
-
+  private static final Pattern PATTERN_MGRS_ZONE_AND_LATITUDE_BAND =
+      Pattern.compile(
+          "("
+              + UsngCoordinate.ZONE_REGEX_STRING
+              + ")"
+              + UsngCoordinate.LATITUDE_BAND_PART_ONE_REGEX_STRING);
```

Comment:
❗️ By doing this, we are tightly coupling ourselves to the USNG libraries. There is still discussion to be had if the fuzzy & forgiving text parsing aspect belongs at that low level. If we feel it does, we still haven't proven how reusable that might be. Nor its long term stability. 

USNG is about coordinate conversions. Not necessarily text parsing. It's much easier and safer in scope to limit the change to the regexs here without coupling to an implementation detail in the library. 

Thoughts?

Reply:
Re-implementing the same piece of functionality in two different places is hard to maintain, the purpose of USNG library is to provide conversion between coordinate types and for parsing those coordinate values from strings.  We actually use the USNG in this class, it is not being added; the only change is the use of UNSG library is now more consistent and the code/functionality duplication is removed.

Reply:
I guess you could make the argument these are constant strings, exposed by the library, that represent living documentation about the coordinate spec itself. There is no guarantee that the usng4j impl itself is actually using these values. 

If we're okay with this, then I am too 👍